src/modules/company/company.service.ts

```diff
@@ -58,4 +84,41 @@ export class CompanyService {
     detele(id: string) {
         return this.prismaService.company.delete({ where: { id } })
     }
+
+    buildSearchParams(
+        query: string,
+        filters: CompanySearchFilterDTO,
+        mode: Prisma.QueryMode,
+    ): Prisma.CompanyWhereInput {
+        const data: Prisma.CompanyWhereInput = {
+            OR: [
+                {
+                    name: { contains: query, mode: mode },
+                    description: {
+                        contains: query,
+                        mode: mode,
+                    },
+                    shortDescription: {
+                        contains: query,
+                        mode: mode,
+                    },
+                    companyName: {
+                        contains: query,
+                        mode: mode,
+                    },
+                    state: { contains: query, mode: mode },
+                    city: { contains: query, mode: mode },
+                },
+            ],
+
+            AND: [
+                {
+                    published: filters.published,
+                    userCompanyRelation: { some: { userId: filters.userId } },
```

Comment:
Acho que não precisa do `some`
```suggestion
                    userCompanyRelation: { userId: filters.userId },
```